docs/reference/remap.cue

```diff
@@ -65,4 +64,205 @@ remap: {
+				"""
+			use: ["parameter", "return"]
+		}
+		"timestamp": {
+			description: "A string formatted as a timestamp."
+			use: ["parameter", "return"]
+		}
+	}
+
+	// TRL syntax
+	#Operators: [_category=string]: [_op=string]: string
+
+	syntax: [RuleName=string]: {
+		name:        RuleName
+		href:        string // Ensures that we don't end up with clashing anchors
+		description: string
+		examples: [string, ...string]
+		operators?: #Operators
+	}
+
+	syntax: {
+		"Dot notation": {
+			href: "dot-notation"
+
+			description: """
+				In TRL, a dot (`.`) holds state across the script. At the beginning of the script,
+				it represents the event arriving into the transform. Take this JSON event data as an
+				example:
+
+				```json
+				{"status_code":200,"username":"booper1234","message":"Successful transaction"}
+				```
+
+				In this case, the event, represented by the dot, has three fields: `.status_code`,
+				`.username`, and `.message`. You can assign new values to the existing fields
+				(`.message = "something different"`), add new fields (`.new_field = "new value"`),
+				delete fields (`del(.username)`), store those values in variables (`$code =
+				.status_code`), and more.
+
+				The dot can also represent nested values, for example `.transaction.id` or
+				`.geo.latitude`.
+				"""
+			examples: [
+				".",
+				".status_code",
+				".message",
+				".username",
+				".transaction.id",
+				".geo.latitude",
+			]
+		}
+
+		"Functions": {
+			href: "functions"
+
+			description: """
+				In TRL, functions act just like they do in standard programming languages. They can
+				take inputs (or no input) and either return values or make an
+				[assertion](#assertion) using the [`assert`](#assert) function.
+
+				TRL types that can serve as inputs to functions are listed in [Parameter
+				types](#parameter-types); types that functions can output are listed in [Return
+				types](#return-types).
+				"""
+
+			examples: [
+				"parse_json(.message)",
+				"assert(.status_code == 500)",
+				#"ip_subnet(.address, "255.255.255.0")"#,
+			]
+		}
+
+		"Assignment": {
+			href: "assignment"
+
+			description: """
+				You can assign values to fields using a single equals sign (`=`). If the field
+				already exists, its value is re-assigned; it the field doesn't already exist, it's
+				created and assigned the value.
+				"""
+
+			examples: [
+				".request_id = uuidv4()",
+				".average = .total / .number",
+				".partition_id = .status_code",
+				".is_server_error = .status_code == 500",
+			]
+		}
+
+		"Assertion": {
+			href: "assertion"
+
+			description: """
+				In TRL, you can assert that something is true using [`assert`](#assert). If an
+				assertion succeeds, Vector does nothing and continues with the script; if it fails,
+				the script stops executing and an error is logged. `assert` can take any Boolean as
+				input, including functions that return Booleans, like [`exist`](#exist).
+				"""
```

Comment:
Similar to above, I'm not entirely convinced this function should be part of the syntax section. It's "just" a function, and one that might not be available in all contexts (as with all functions, it's injected into a TRL runtime, and by default TRL has 0 functions).